File: Weapon/Weapon.py
from Money import Money


class Weapon:
    def __init__(self, price: Money, name: str):
        self.__name = name
        self.__price = price

    @property
    def name(self):
        return self.__name

    @property
    def price(self):
        return self.__price.amount

    @price.setter
    def price(self, value: Money):
        self.__price = value

File: Weapon/Weapon.php
<?php

namespace Weapon;

abstract class Weapon
{
    private $name;
    private $price;

    final public function __construct(float $price,string $name)
    {
        $this->name =$name;
        $this->price = $price;
    }

    final public function getPrice(){
        return $this->price;
    }
    final public function getName(){
        return $this->name;
    }

}